submission/submitter/src/test/kotlin/ac/uk/ebi/biostd/submission/validator/collection/EuToxRiskValidatorTest.kt

```diff
@@ -0,0 +1,99 @@
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.core.io.FileSystemResource
+import org.springframework.http.HttpEntity
+import org.springframework.web.client.RestTemplate
+
+@ExtendWith(MockKExtension::class, TemporaryFolderExtension::class)
+class EuToxRiskValidatorTest(
+    private val temporaryFolder: TemporaryFolder,
+    @MockK private val restTemplate: RestTemplate,
+    @MockK private val validationProperties: ValidatorProperties
+) {
+    private val testUrl = "http://eutoxrisk.org/validator"
+    private val textFile = temporaryFolder.createFile("test.txt")
+    private val excelFile = temporaryFolder.createFile("test.xlsx")
+    private val testInstance = EuToxRiskValidator(restTemplate, validationProperties)
+
+    @BeforeEach
+    fun beforeEach() {
+        every { validationProperties.euToxRiskValidationApi } returns testUrl
+    }
+
+    @AfterEach
+    fun afterEach() = clearAllMocks()
+
+    @Test
+    fun validate() {
+        val requestSlot = setUpRestTemplate(EuToxRiskValidatorResponse(listOf()))
+        val submission = basicExtSubmission.copy(section = ExtSection(
+            type = "Study",
+            files = listOf(left(ExtFile("test.xlsx", excelFile)))
+        ))
+
+        testInstance.validate(submission)
+
+        verify(exactly = 1) {
+            restTemplate.postForObject(testUrl, requestSlot.captured, EuToxRiskValidatorResponse::class.java)
+        }
+    }
+
+    @Test
+    fun `validate with errors`() {
+        val submission = basicExtSubmission.copy(section = ExtSection(
+            type = "Study",
+            files = listOf(left(ExtFile("test.xlsx", excelFile)))
+        ))
+        val requestSlot = setUpRestTemplate(EuToxRiskValidatorResponse(listOf(EuToxRiskValidatorMessage("an error"))))
+
+        val error = assertThrows<CollectionValidationException> { testInstance.validate(submission) }
+        assertThat(error.message).isEqualTo(
+            "The submission doesn't comply with the collection requirements. Errors: [an error]")
+        verify(exactly = 1) {
+            restTemplate.postForObject(testUrl, requestSlot.captured, EuToxRiskValidatorResponse::class.java)
+        }
+    }
+
+    @Test
+    fun `submission without excel file`() {
+        val submission = basicExtSubmission.copy(section = ExtSection(
+            type = "Study",
+            files = listOf(left(ExtFile("test.txt", textFile)))
+        ))
+
+        val error = assertThrows<CollectionValidationException> { testInstance.validate(submission) }
+        assertThat(error.message).isEqualTo(
+            "The submission doesn't comply with the collection requirements. Errors: [$EXCEL_FILE_REQUIRED]")
+    }
+
+    private fun setUpRestTemplate(response: EuToxRiskValidatorResponse): CapturingSlot<HttpEntity<FileSystemResource>> {
+        val requestSlot = slot<HttpEntity<FileSystemResource>>()
```

Comment:
IMO we should declare this as a field and in each test:
```
  every {  restTemplate.postForObject<>(testUrl, capture(requestSlot)) } } returns response
```
I think way when one is reviewing the test does not need to check what `setUpRestTemplate` does as the setup is part of each test. Also I think we do not need type  I think compiler with infer it.